sdk/program/src/vote/state/mod.rs

```diff
@@ -818,6 +873,60 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_vote_deserialize_into() {
+        // base case
+        let target_vote_state = VoteState::default();
+        let vote_state_buf =
+            bincode::serialize(&VoteStateVersions::new_current(target_vote_state.clone())).unwrap();
+
+        let mut test_vote_state = VoteState::default();
+        VoteState::deserialize_into(&vote_state_buf, &mut test_vote_state).unwrap();
+
+        assert_eq!(target_vote_state, test_vote_state);
+
+        // variant
+        for _ in 0..1000 {
+            let size_hint = VoteStateVersions::size_hint(0).0 * 4;
+            let raw_data: Vec<u8> = (0..size_hint).map(|_| rand::random::<u8>()).collect();
+            let mut unstructured = Unstructured::new(&raw_data);
+
+            let target_vote_state_versions =
+                VoteStateVersions::arbitrary(&mut unstructured).unwrap();
+            let vote_state_buf = bincode::serialize(&target_vote_state_versions).unwrap();
+            let target_vote_state = target_vote_state_versions.convert_to_current();
+
+            let mut test_vote_state = VoteState::default();
+            VoteState::deserialize_into(&vote_state_buf, &mut test_vote_state).unwrap();
+
+            assert_eq!(target_vote_state, test_vote_state);
+        }
+    }
+
+    #[test]
+    fn test_vote_deserialize_into_nopanic() {
+        // base case
+        let mut test_vote_state = VoteState::default();
+        let e = VoteState::deserialize_into(&[], &mut test_vote_state).unwrap_err();
+        assert_eq!(e, InstructionError::InvalidAccountData);
+
+        // variant
+        let seed = rand::random::<u64>();
+        msg!("test_vote_deserialize_into_nopanic seed: {}", seed);
+        let mut r = StdRng::seed_from_u64(seed);
+
+        for _ in 0..1000 {
+            let raw_data_length = r.gen_range(1..1024);
```

Comment:
Where does this number come from?

Reply:
its arbitrary, i could note this in a comment?

Reply:
going to use the serialized size of the default struct instead